src/andromede/simulation/optimization.py

```diff
@@ -679,34 +668,22 @@ def make_constraint(
 
 
 class OptimizationProblem:
-    class Type(Enum):
-        """
-        Class to specify the type of the created problem:
-            - master: Creates a Xpansion master problem with investment and coupling variables and constraints only
-            - subproblem: Creates Xpansion sub-problems with operational and coupling variables and constraints only
-            - merged: Creates a Antares Simulator/ Xpansion problem with all
-        """
-
-        merged = 0
-        master = 1
-        subproblem = 2
-
     name: str
     solver: lp.Solver
     context: OptimizationContext
-    problem_type: Type
+    strategy: Type[ModelSelectionStrategy]
```

Comment:
`ModelSelectionStrategy` is not enough for typing ?

Reply:
Unfortunately no, because it can also represent the child classes, not only ```ModelSelectionStrategy```

Reply:
Okay I didn't known that !